### ISSUE TITLE
SQLAlchemy trying to parse an azure connectionstring and failing

### ISSUE THREAD

**Issue by c624ad2** [great-expectations/great_expectations#7997]
**Describe the bug**
I have a situation where I store my expectations and validation results in an azure blob. To be able to do this I have defined an azure-connectionstring and store it in a key vault. When I generate/validate expectations I write extract the connectionstring from azure and then write it to `config_variables `and when the connection is closed I delete the connectionstring from the yaml file. 

The error message: 
Something went wrong when trying to use SQLAlchemy to obfuscate URL: Could not parse rfc1738 URL from string

I have traced the problem and it occurs in the method [great_expectations.util.PasswordMasker.sanitize_config](https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/data_context/util.py#L254).

**To Reproduce**
I have created a quickfix in my own local copy of great_expectations to solve this:

Existing code:
```py
def sanitize_config(cls, config: dict) -> dict:
        """
        Mask sensitive fields in a Dict.
        """

        # be defensive, since it would be logical to expect this method works with DataContextConfig
        if not isinstance(config, dict):
            raise TypeError(
                "PasswordMasker.sanitize_config expects param `config` "
                + f"to be of type Dict, not of type {type(config)}"
            )

        config_copy = safe_deep_copy(config)  # be immutable

        def recursive_cleaner_method(config: Any) -> None:
            if isinstance(config, dict):
                for key, val in config.items():
                    if not isinstance(val, str):
                        recursive_cleaner_method(val)
                    elif key in cls.URL_KEYS:
                        config[key] = cls.mask_db_url(val)
                    elif key in cls.PASSWORD_KEYS:
                        config[key] = cls.MASKED_PASSWORD_STRING
                    else:
                        pass  # this string is not sensitive
            elif isinstance(config, list):
                for val in config:
                    recursive_cleaner_method(val)

        recursive_cleaner_method(config_copy)  # Perform anonymization in place

        return config_copy
```
My local quickfix:
```py
def sanitize_config(cls, config: dict) -> dict:
        """
        Mask sensitive fields in a Dict.
        """

        # be defensive, since it would be logical to expect this method works with DataContextConfig
        if not isinstance(config, dict):
            raise TypeError(
                f"PasswordMasker.sanitize_config expects param `config` "
                + f"to be of type Dict, not of type {type(config)}"
            )

        config_copy = safe_deep_copy(config)  # be immutable

        # Define a regex to recognize azure storage connection strings
        azure_conn_str_re = re.compile(
            "(DefaultEndpointsProtocol=(http|https));(AccountName=([a-zA-Z0-9]+));(AccountKey=)([a-zA-Z0-9=\+]+);(EndpointSuffix=([a-zA-Z\.]+))"
        )

        def recursive_cleaner_method(config: Any) -> None:
            if isinstance(config, dict):
                for key, val in config.items():
                    if not isinstance(val, str):
                        recursive_cleaner_method(val)
                    elif key in cls.URL_KEYS:
                        # Special case if the connection string is for azure storage
                        if key == "connection_string" and azure_conn_str_re.match(val):
                            # Get the match parts
                            m = azure_conn_str_re.match(val)
                            # Update the connection_string while masking the password with ***
                            config[
                                key
                            ] = f"DefaultEndpointsProtocol={m.group(2)};AccountName={m.group(4)};AccountKey=***;EndpointSuffix={m.group(7)}"
                        else:
                            config[key] = cls.mask_db_url(val)
                    elif key in cls.PASSWORD_KEYS:
                        config[key] = cls.MASKED_PASSWORD_STRING
                    else:
                        pass  # this string is not sensitive
            elif isinstance(config, list):
                for val in config:
                    recursive_cleaner_method(val)

        recursive_cleaner_method(config_copy)  # Perform anonymization in place

        return config_copy
```

**Expected behavior**
It would be nice if this error could be avoided by either correctly sanitizing azure connectionstrings or ignoring them.

**Environment (please complete the following information):**
 - Operating System: Windows
 - Great Expectations Version: 0.15.3 (but this was an issue in earlier versions as well.

**Additional context**
I couldn't find any other suggestions online of how to solve this, so thought that an issue would be best in case anyone else has experienced it as well:)

**Comment by tjholsman** [comment]
Thanks for logging this, @c624ad2! Would you be interested in submitting your fix as a PR for consideration? There is more information in our [contributing doc](https://github.com/great-expectations/great_expectations/blob/develop/CONTRIBUTING_CODE.md).

**Comment by tjholsman** [comment]
@c624ad2 thanks for creating the PR! Unfortunately it looks like this PR was made on your forked repo and not on the GX repo. 

Here's our contributing guide: https://github.com/great-expectations/great_expectations/blob/develop/CONTRIBUTING_CODE.md

And this guide from GitHub on creating a PR from a fork (linked within our guide): https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork

When done correctly, you'll be prompted to complete our CLA and then our CI checks will run automatically.

Could you make these adjustments when you have a moment?

**Comment by c624ad2** [comment]
@tjholsman I have made a new PR now with the checks!

**Comment by tyler-hoffman** [comment]
I left a note on the PR, but this was resolved by https://github.com/great-expectations/great_expectations/pull/7139, and was released in gx version 0.15.49 I'm closing this ticket out, but please reopen if you run into any issues after upgrading. Thanks!
